From 4a38711a2777b1fbe1a043eca40d9c741021ed4c Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Wed, 15 Aug 2018 10:12:59 +0700
Subject: [PATCH] quick: scenegraph: fix texture width alignment issue for
 PowerVR G6400

This commit fix an  unclear crash issue when using PowerVR G6400 on iwg21m board

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 src/quick/scenegraph/qsgdefaultdistancefieldglyphcache.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/quick/scenegraph/qsgdefaultdistancefieldglyphcache.cpp b/src/quick/scenegraph/qsgdefaultdistancefieldglyphcache.cpp
index 31275b3..fadf381 100644
--- a/src/quick/scenegraph/qsgdefaultdistancefieldglyphcache.cpp
+++ b/src/quick/scenegraph/qsgdefaultdistancefieldglyphcache.cpp
@@ -100,6 +100,8 @@ void QSGDefaultDistanceFieldGlyphCache::requestGlyphs(const QSet<glyph_t> &glyph
         QRectF boundingRect = glyphData(glyphIndex).boundingRect;
         int glyphWidth = qCeil(boundingRect.width()) + distanceFieldRadius() * 2;
         int glyphHeight = qCeil(boundingRect.height()) + distanceFieldRadius() * 2;
+        //workaround to prevent odd width issue  when using PowerVR Rogue G6400
+        glyphWidth += glyphWidth % 2;
         QSize glyphSize(glyphWidth + padding * 2, glyphHeight + padding * 2);
         QRect alloc = m_areaAllocator->allocate(glyphSize);

--
2.7.4
